README.md

```diff
@@ -1,34 +1,84 @@
+[![Hex.pm](https://img.shields.io/hexpm/v/membrane_ffmpeg_swscale_plugin.svg)](https://hex.pm/packages/membrane_ffmpeg_swscale_plugin)
+[![API Docs](https://img.shields.io/badge/api-docs-yellow.svg?style=flat)](https://hexdocs.pm/membrane_ffmpeg_swscale_plugin/)
+[![CircleCI](https://circleci.com/gh/membraneframework/membrane_ffmpeg_swscale_plugin.svg?style=svg)](https://circleci.com/gh/membraneframework/membrane_ffmpeg_swscale_plugin)
 
-This repository contains a template for new plugins.
+Plugin providing an element scaling raw video frames, using SWScale module of [FFmpeg](https://www.ffmpeg.org/) library.
 
-Check out different branches for other flavours of template.
-
-It is part of [Membrane Multimedia Framework](https://membraneframework.org).
+It is a part of [Membrane Multimedia Framework](https://membraneframework.org).
 
 ## Installation
 
-The package can be installed by adding `membrane_template_plugin` to your list of dependencies in `mix.exs`:
+Add the following line to your `deps` in `mix.exs`. Run `mix deps.get`.
 
 ```elixir
-def deps do
-  [
-    {:membrane_template_plugin, "~> 0.1.0"}
-  ]
-end
+{:membrane_ffmpeg_swscale_plugin, "~> 0.1.0"}
 ```
 
-## Usage
+You also need to have [FFmpeg](https://www.ffmpeg.org/) library with development headers installed on your system.
+
+## Description
+
+Scaler needs input in the YUV420p format, processes one frame at a time and requires getting caps with input video
+width and height. To meet all requirements either `Membrane.Element.RawVideo.Parser` or some decoder
+(e.g. `Membrane.H264.FFmpeg.Decoder`) have to precede Scaler in the pipeline.
+
+There are two options that have to be specified when creating the element:
+- `output_width` - desired scaled video width.
+- `output_height` - desired scaled video height.
+
+Both need to be even numbers.
+
+Scaling consists of two operations:
+- scaling itself - resizing video frame with keeping original ratio. After that operation at least one of the
+dimensions of the input frame match the respective dimension of the desired output size. The second one
+(if does not match) is smaller than its respective dimension.
+- adding paddings - if one dimension does not match after scaling, paddings have to be added. They are put on both
+sides of the scaled frame equally. They are either above and below the frame or on the left and right sides of it.
+It depends on the dimension that did not match after scaling.
+
+The output of the element is also in the YUV420p format. It has the size as specified in the options. All
+caps except for width and height are passed unchanged to the next element in the pipeline.
 
-TODO
+## Sample usage
+
+Scaling encoded (using H.264 standard) video requires parser and decoder because Scaler scales only raw video.
+
+```elixir
+defmodule Scaling.Pipeline do
+  use Membrane.Pipeline
+
+  @doc false
```

Comment:
```suggestion
```
It's not needed when using `@impl true`

On a side note, be careful when writting `@impl`, `@doc` etc. in comments, you should always wrap them backticks - otherwise you'll notify someone using doc or impl as GitHub handle